src/transformers/models/bloom/parallel_layers.py

```diff
@@ -0,0 +1,235 @@
+import torch
+import torch.distributed
+import torch.nn.functional as F
+from torch import nn
+import math
+
+
+class TensorParallelColumnLinear(nn.Module):
+    def __init__(
+        self,
+        in_features,
+        out_features,
+        process_group: torch.distributed.ProcessGroup,
+        bias=True,
+        device=None,
+        dtype=None,
+    ):
+        factory_kwargs = {'device': device, 'dtype': dtype}
+        super().__init__()
+
+        self.process_group = process_group
+        self.tp_world_size = process_group.size()
+
+        assert out_features % self.tp_world_size == 0
+
+        self.in_features = in_features
+        self.out_features = out_features // self.tp_world_size
+        # We change from traditional `nn.Linear` and remove unecessary `torch.Tensor.transpose` operation
```

Comment:
I think this adds an unecessary overhead. Let's make those class simple, and inherit `nn.Linear`. We'll then be able to remove `reset_parameters` and `extra_repr` that I've shamelessly copied.

Reply:
Let's keep this in another PR then.

Reply:
Or take care of it if you want, but I don't really want to spend too much time on correctness here (and I know the current implem works since it's used in prod)